core/http-auth-spi/src/test/java/software/amazon/awssdk/http/auth/spi/HttpSignerTest.java

```diff
@@ -0,0 +1,113 @@
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.spi;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+
+import java.nio.ByteBuffer;
+import org.junit.jupiter.api.Test;
+import org.reactivestreams.Publisher;
+import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.identity.spi.TokenIdentity;
+
+public class HttpSignerTest {
+
+    private static final SignerProperty<String> KEY = SignerProperty.create(String.class, "key");
+    private static final String VALUE = "value";
+    private TokenIdentity IDENTITY = TokenIdentity.create("token");
+
+    final HttpSigner<TokenIdentity> signer = new TestSigner();
+
+    @Test
+    public void sign_usingConsumerBuilder_works() {
+        SignedHttpRequest<ContentStreamProvider> signedRequest = signer.sign(r -> r.request(mock(SdkHttpRequest.class))
+                                                                                   .identity(IDENTITY)
+                                                                                   .putProperty(KEY, VALUE));
+        assertNotNull(signedRequest);
+    }
+
+    @Test
+    public void sign_usingRequest_works() {
+        SignedHttpRequest<ContentStreamProvider> signedRequest =
+            signer.sign(HttpSignRequest.builder(ContentStreamProvider.class, TokenIdentity.class)
+                                                 .request(mock(SdkHttpRequest.class))
+                                                 .identity(IDENTITY)
+                                                 .putProperty(KEY, VALUE)
+                                                 .build());
+        assertNotNull(signedRequest);
+    }
+
+    @Test
+    public void signAsync_usingConsumerBuilder_works() {
+        Publisher<ByteBuffer> payload = subscriber -> {};
+        SignedHttpRequest<Publisher<ByteBuffer>> signedRequest = signer.signAsync(r -> r.request(mock(SdkHttpRequest.class))
+                                                                                        .payload(payload)
+                                                                                        .identity(IDENTITY)
+                                                                                        .putProperty(KEY, VALUE));
+        assertNotNull(signedRequest);
+    }
+
+    // @Test
+    public void signAsync_usingRequest_works() {
+        Publisher<ByteBuffer> payload = subscriber -> {};
+        // Don't know how to make this work....
+        // HttpSignRequest.Builder<Publisher<ByteBuffer>, TokenIdentity> builder =
+        //   HttpSignRequest.builder(Publisher.class, TokenIdentity.class);
```

Comment:
I'm unsure what would this have looked like without the Consumer builder pattern? This would have been a question without making these changes to pull Identity out explicitly.

Reply:
Resolved by moving to separate sub types for sync and async.